squashfs/squashfs.go

```diff
@@ -175,7 +175,14 @@ func ExtractSingleSquash(squashFile string, extractDir string, storageType strin
 		return which("squashfuse")
 	}
 
-	if sqfuse := findSqfusePath(); sqfuse != "" {
+	if realuid := os.Getenv("STACKER_REAL_UID"); realuid == "0" {
```

Comment:
If I'm running stacker inside an unprivileged container as root, my STACKER_REAL_UID will be 0 right?

What if we ... just try it?  Try the mount -o loop, if cmd.Run() fails with EPERM, then (a) log that and don't try again, and (b) try squashfuse.